File: app/src/main/java/com/maxwellscore/sparkone/data/dto/TemperatureData.java
package com.maxwellscore.sparkone.data.dto;

import com.google.gson.annotations.SerializedName;
import com.maxwellscore.sparkone.data.DataConstants;

public class TemperatureData {

    @SerializedName(DataConstants.KEY_TEMPERATURE)
    private double temperature;
    @SerializedName(DataConstants.KEY_UNIT)
    private String unit;

    public TemperatureData() {
    }

    public TemperatureData(double temperature, String unit) {
        this.temperature = temperature;
        this.unit = unit;
    }

    public double getTemperature() {
        return temperature;
    }

    public String getUnit() {
        return unit;
    }
}

File: kotlinexample/src/main/java/com/maxwellscore/kotlinexample/data/dto/TemperatureData.kt
package com.maxwellscore.kotlinexample.data.dto

import com.google.gson.annotations.SerializedName
import com.maxwellscore.kotlinexample.data.DataConstants

data class TemperatureData(
    @SerializedName(DataConstants.KEY_TEMPERATURE)
    val temperature: Int,
    @SerializedName(DataConstants.KEY_UNIT)
    val unit: String
)
